samples/local_attestation/common/dispatcher.cpp

```diff
@@ -95,62 +95,47 @@ int ecall_dispatcher::get_target_info(
 
     // Generate a report for the public key so that the enclave that
     // receives the key can attest this enclave.
-    report = (uint8_t*)oe_host_malloc(OE_MAX_REPORT_SIZE);
-    if (report == NULL)
-    {
-        goto exit;
-    }
-    report_size = OE_MAX_REPORT_SIZE;
-
     if (m_attestation->generate_local_report(
-            NULL, 0, NULL, 0, report, &report_size))
+            NULL, 0, NULL, 0, &report, &report_size))
     {
         size_t info_size = 0;
 
         TRACE_ENCLAVE("report_size = %ld", report_size);
         // get the target info
-        result =
-            oe_get_target_info(report, report_size, info_buffer, &info_size);
-        if (result != OE_BUFFER_TOO_SMALL)
+        // caller of info_buffer needs to free this memory
+        result = oe_get_target_info(
+            report, report_size, (void**)&info_buffer, &info_size);
```

Comment:
What's going on here `(void**)&info_buffer`?

Reply:
The prototype:
https://github.com/Microsoft/openenclave/blob/master/include/openenclave/enclave.h#L383

I followed the guidelines from the tests:
https://github.com/Microsoft/openenclave/blob/master/tests/report/common/tests.cpp#L587
Please let me know if you want me to change it.

Reply:
Sounds good to me.